tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php

```diff
@@ -195,4 +195,17 @@ private function assertLimitResult(
             self::assertCount(count($expectedResults), $data);
         }
     }
+
+    public function testLimitWhenOrderByWithLimitedSubquery(): void
+    {
+        $this->connection->insert('modify_limit_table2', ['test_int' => 1]);
+        $this->connection->insert('modify_limit_table2', ['test_int' => 2]);
+
+        $subquery        = 'SELECT test_int FROM modify_limit_table2';
+        $p               = $this->connection->getDatabasePlatform();
+        $limitedSubquery = $p->modifyLimitQuery($subquery, 1, 0);
+        $sql             = 'SELECT test_int FROM modify_limit_table2 ORDER BY (' . $limitedSubquery . ') ASC';
```

Comment:
It doesn't look like the case you originally tried to implement. The one in the unit test looks quote generic and should work with most of the platforms while this one is quite exotic.